src/qengine/state/operators.cpp

```diff
@@ -834,41 +836,36 @@ bitCapInt QEngineCPU::MReg(bitLenInt start, bitLenInt length)
         NormalizeState();
     }
 
-    bool foundPerm;
     double prob = Rand();
     double angle = Rand() * 2.0 * M_PI;
     double cosine = cos(angle);
     double sine = sin(angle);
     bitCapInt lengthPower = 1 << length;
     bitCapInt regMask = (lengthPower - 1) << start;
-    double probArray[lengthPower];
-    double lowerProb, largestProb, nrmlzr;
+    double* probArray = new double[lengthPower]();
```

Comment:
Unnecessary () at the end.

Reply:
This isn't unnecessary. It's the "calloc" constructor for a primitive. It indicates that an array of primitives should be initialized to their default (0.0) state.